lua/oil/config.lua

```diff
@@ -408,6 +416,9 @@ M.setup = function(opts)
   if opts.preview and not opts.confirmation then
     new_conf.confirmation = vim.tbl_deep_extend("keep", opts.preview, default_config.confirmation)
   end
+
+  new_conf.new_file_mode = tonumber(tostring(new_conf.new_file_mode), 8)
+  new_conf.new_dir_mode = tonumber(tostring(new_conf.new_dir_mode), 8)
```

Comment:
nit: you don't actually need the `tostring()` here. `tonumber` will convert an integer or a string the same way.

Reply:
Not sure about it, i try the following code:
```lua
local x = 493
print(tonumber(x, 8))
```
and I get the following error:
```
$ lua test.lua
lua: test.lua:2: bad argument #1 to 'tonumber' (string expected, got number)
stack traceback:
        [C]: in function 'tonumber'
        test.lua:2: in main chunk
        [C]: in ?
```
I'm not a Lua expert, in fact this is my first productive code in Lua, so maybe I'm ignoring something

Reply:
the second argument to [tonumber](https://www.lua.org/manual/5.1/manual.html#pdf-tonumber) is the base to interpret the number as. 493 is not a valid octal number. If you do `tonumber(755, 8)` it prints out `493` as expected

Reply:
Again the same:
```
$ lua -e "print(tonumber(755, 8))"
lua: (command line):1: bad argument #1 to 'tonumber' (string expected, got number)
stack traceback:
        [C]: in function 'tonumber'
        (command line):1: in main chunk
        [C]: in ?
```
Maybe this has something to do with my interpreter? to run this examples I used the standalone interpreter:
```
$ lua
Lua 5.4.6  Copyright (C) 1994-2023 Lua.org, PUC-Rio
```

Reply:
Oh interesting, must be a difference between LuaJIT and Lua then (or 5.1 and 5.4). I retract the nit